internal/services/elasticsan/validate/elastic_san_name.go

```diff
@@ -0,0 +1,43 @@
+package validate
+
+import (
+	"fmt"
+	"regexp"
+)
+
+func ElasticSanName(i interface{}, k string) (warnings []string, errors []error) {
+	v, ok := i.(string)
+	if !ok {
+		errors = append(errors, fmt.Errorf("expected %q to be a string but it wasn't!", k))
+		return
+	}
+
+	// name can be 3-24 characters in length
+	const minLength = 3
```

Comment:
Is it possible to combine all or some of these conditions into one? like https://github.com/hashicorp/terraform-provider-azurerm/blob/183cd43e6dc7b6855ecddecc114c3c441f3cc4ee/internal/services/machinelearning/machine_learning_compute_instance_resource.go#L53

Reply:
Hard, for I don't know how to integrate `must have hyphens and underscores be surrounded by alphanumeric character` in one regex.

Reply:
Expect this one, can we combine other parts into one?